contracts/managers/withdraw/WithdrawManager.sol

```diff
@@ -48,42 +79,50 @@ contract WithdrawManager is AccessManagedUpgradeable, ContextClient {
                 request.endWithdrawWindowTimestamp,
                 WithdrawManagerStorageLib.getWithdrawWindowInSeconds(),
                 releaseFundsTimestamp
-            ) && request.amount >= amount_
+            ) && request.shares >= shares_
         ) {
-            WithdrawManagerStorageLib.deleteWithdrawRequest(account_, amount_);
+            WithdrawManagerStorageLib.decreaseSharesFromWithdrawRequest(account_, shares_);
+            WithdrawManagerStorageLib.decreaseSharesToRelease(shares_);
             return true;
         }
-
-        uint256 balanceOfPlasmaVault = ERC4626(ERC4626(msg.sender).asset()).balanceOf(msg.sender);
-        uint256 amountToRelease = WithdrawManagerStorageLib.getAmountToRelease();
-
-        if (balanceOfPlasmaVault >= amountToRelease) {
-            return balanceOfPlasmaVault - amountToRelease >= amount_;
-        }
         return false;
     }
 
-    /**
-     * @notice Creates a new withdrawal request
-     * @dev Publicly accessible function
-     * @param amount_ The amount requested for withdrawal
-     * @custom:access Public
-     */
-    function request(uint256 amount_) external {
-        WithdrawManagerStorageLib.updateWithdrawRequest(_msgSender(), amount_);
+    function canWithdrawFromUnallocated(uint256 shares_) external restricted returns (uint256 feeSharesToBurn) {
```

Comment:
Add description to this method.  Try to explain what unallocated means in this context